### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/knot-resolver/commit/7bcf46a21800f0c076391085ed465d006cc875bb#commitcomment-60441394

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/knot-resolver/commit/5bab6477bfa465013300598e1c87449bb4e97fa8

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ variables:
   LIBKRES_NAME: libkres
   MESON_TEST: meson test -C build_ci* -t 4 --print-errorlogs
   PREFIX: $CI_PROJECT_DIR/.local
-  GIT_COMMITER_NAME: 'ci'
```
Comment (jsoref) [inline .gitlab-ci.yml:18]: ?

File: bench/bench_lru.c [hunk 2]
```diff
@@ -143,7 +143,7 @@ static struct key * read_lines(const char *fname, size_t *count, char **pfree)
 	return result;
 }
 
-// compatibility layer for the oler lru_* names; it's more compler with lru_create
+// compatibility layer for the older lru_* names; it's more complete with lru_create
```
Comment (jsoref) [inline bench/bench_lru.c:146]: I couldn't guess for the second word here

File: doc/conf.py [hunk 3]
```diff
@@ -59,11 +59,11 @@
 
 html_theme_options = {
     'logo_only': True,  # if we have a html_logo below, this shows only the logo with no title text
-    # Toc options
+    # ToC options
     'collapse_navigation': False,
     'sticky_navigation': True,
 }
-html_logo = '_static/logo-negativ.svg'
```
Comment (jsoref) [inline doc/conf.py:66]: This file is renamed.

Note: if a given family of changes is objectionable, it's trivial for me to drop, just let me know.

File: lib/log.h [hunk 4]
```diff
@@ -96,15 +96,15 @@ enum kr_log_group {
...
+#define LOG_GRP_TLSCLIENT_TAG		"tls_cl"	/**< ``tls_cl``: TLS client messages (used for TLS forwarding) */
 #define LOG_GRP_XDP_TAG			"xdp"		/**< ``xdp``: operations related to XDP */
 #define LOG_GRP_ZIMPORT_TAG		"zimprt"	/**< ``zimprt``: operations related to zimport */
 #define LOG_GRP_ZSCANNER_TAG		"zscann"	/**< ``zscann``: operations related to zscanner */
 #define LOG_GRP_DOH_TAG			"doh"		/**< ``doh``: DNS-over-HTTPS logger (doh2 implementation) */
 #define LOG_GRP_DNSSEC_TAG		"dnssec"	/**< ``dnssec``: operations related to DNSSEC */
 #define LOG_GRP_HINT_TAG		"hint"		/**< ``hint``: operations related to static hints */
 #define LOG_GRP_PLAN_TAG		"plan"		/**< ``plan``: operations related to resolution plan */
-#define LOG_GRP_ITERATOR_TAG		"iterat"	/**< ``iterat``: operations related to iterate layer */
+#define LOG_GRP_ITERATOR_TAG		"iterate"	/**< ``iterate``: operations related to iterate layer */
```
Comment (jsoref) [inline lib/log.h:107]: I meant to avoid changing this table

File: lib/log.h [hunk 5]
```diff
@@ -127,7 +127,7 @@ enum kr_log_group {
 #define LOG_GRP_TESTS_TAG		"tests"		/**< ``tests``: operations related to tests  */
 #define LOG_GRP_DOTAUTH_TAG		"dotaut"	/**< ``dotaut``: DNS-over-TLS against authoritative servers */
 #define LOG_GRP_HTTP_TAG		"http"		/**< ``http``: http module, its web interface and legacy DNS-over-HTTPS */
-#define LOG_GRP_CONTROL_TAG		"contrl"	/**< ``contrl``: TTY control sockets*/
+#define LOG_GRP_CONTROL_TAG		"control"	/**< ``control``: TTY control sockets*/
```
Comment (jsoref) [inline lib/log.h:130]: I meant to avoid changing this table

File: lib/resolve.c [hunk 6]
```diff
@@ -207,18 +207,18 @@ static int ns_fetch_cut(struct kr_query *qry, const knot_dname_t *requested_name
 			struct kr_request *req, knot_pkt_t *pkt)
 {
 	/* It can occur that here parent query already have
-	 * provably insecured zonecut which not in the cache yet. */
+	 * provably unsecured zonecut which not in the cache yet. */
```
Comment (jsoref) [inline lib/resolve.c:210]: `insecure`? `unsecure`? `unsecured`?

File: modules/edns_keepalive/edns_keepalive.c [hunk 7]
```diff
@@ -48,7 +48,7 @@ static int edns_keepalive_finalize(kr_layer_t *ctx)
 	return ctx->state;
 }
 
-KR_EXPORT int edns_keeapalive_init(struct kr_module *self)
+KR_EXPORT int edns_keepalive_init(struct kr_module *self)
```
Comment (jsoref) [inline modules/edns_keepalive/edns_keepalive.c:51]: api break

File: modules/http/README.rst [hunk 8]
```diff
@@ -82,7 +82,7 @@ course self-signed. Why not use something like
    If you use package ``luaossl < 20181207``, intermediate certificate is not sent to clients,
    which may cause problems with validating the connection in some cases.
 
-You can disable unecrypted HTTP and enforce HTTPS by passing
+You can disable encrypted HTTP and enforce HTTPS by passing
```
Comment (jsoref) [inline modules/http/README.rst:85]: hmm, probably `unencrypted`

File: tests/config/basic.test.lua [hunk 9]
```diff
@@ -10,7 +10,7 @@ local function test_constants()
 	same(kres.section.ANSWER, 0, 'section constants work')
 	same(kres.rcode.SERVFAIL, 2, 'rcode constants work')
 	same(kres.opcode.UPDATE, 5, 'opcode constants work')
-	-- Test inverset tables to convert constants to text
+	-- Test inverted tables to convert constants to text
```
Comment (jsoref) [inline tests/config/basic.test.lua:13]: ?

File: tests/integration/meson.build [hunk 10]
```diff
@@ -10,7 +10,7 @@ augeas = dependency('augeas')
 # python3 dependencies
 py3_deps += [
   ['augeas', 'augeas (for deckard)'],
-  ['dns', 'dnspuyhon (for deckard)'],
+  ['dns', 'dnspython (for deckard)'],
```
Comment (jsoref) [inline tests/integration/meson.build:13]: ?

**Review by tomaskrizek — CHANGES_REQUESTED** [review]
First of all, thank you! This MR doesn't just fix typos, but you actually discovered and fixed a bug in our edns_keepalive module :tada: 

I think once the individual comments are resolved, it'd be best to squash all typo changes into a single commit - and only leave the EDNS keepalive fix in a separate commit.

This MR got me interested in the check-spelling action and I have a few questions:
- How did you create the [`expect.txt`](https://github.com/jsoref/knot-resolver/commit/7bcf46a21800f0c076391085ed465d006cc875bb#diff-639b39d899fb04589abbb675ef55a5c30df80ae4744ac4b3c6b7f82497473bfa)? Did you have to manually go through all the words and add them one by one?
- We primarily use gitlab, is there a way to use the spell checking tool as a simple command? (I saw the `act` option, but that seem too complex to run a spell check)

File: modules/edns_keepalive/edns_keepalive.c (hunk 7 above)
Comment (jsoref) [reply modules/edns_keepalive/edns_keepalive.c:51]: This doesn't surprise me, it's actually fairly typical IME (and you can imagine I have a lot of experience w/ this) to have 1 or 2 useful finds.

File: modules/policy/policy.lua [hunk 11]
```diff
@@ -621,7 +621,7 @@ function policy.slice_randomize_psl(seed)
 			rand_seed = rand_seed + reg_domain:byte(i)
 		end
 
-		-- use lineral congruential generator with values from ANSI C
+		-- use linearly congruential generator with values from ANSI C
```
[the remark this answers is not in the document]
Comment (jsoref) [reply modules/policy/policy.lua:624]: Sounds like you want `linear` not `lineral`?

File: bench/bench_lru.c (hunk 2 above)
Comment (vcunat) [reply bench/bench_lru.c:146]: I apparently wrote the line, but I'm unable to figure out what I meant, even after understanding (again) what this compatibility layer is about.  I suggest to just remove the part starting at the semicolon.

Anyway, we surely don't even need this compatibility anymore... to micro-benchmark LRU's API that's been gone for over five years.

Comment (vcunat) [reply bench/bench_lru.c:146]: "compatibler" came to my mind, but it would seem really weird.

**Comment by tomaskrizek** [comment]
Changed a couple of things and everything seems good now. Moving this to upstream repository to be merged as https://gitlab.nic.cz/knot/knot-resolver/-/merge_requests/1229

Thanks!

File: doc/conf.py (hunk 3 above)
Comment (vcunat) [reply doc/conf.py]: The name is in Czech, apparently.

**Pull request closed without merging**